AI/scripts/download_weights.py

```
@@ -1,237 +1,421 @@
+
+        if backup_root.exists():
+            shutil.rmtree(backup_root)
+    except Exception as original_exc:
+        rollback_errors: list[str] = []
+        for final_dir, backup_dir in reversed(promoted):
+            try:
+                if final_dir.exists():
+                    shutil.rmtree(final_dir)
+                if backup_dir is not None and backup_dir.exists():
+                    shutil.move(str(backup_dir), str(final_dir))
+            except Exception as rollback_exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)